### ISSUE TITLE
Enable Caliper lifecyle to be controlled more via config file

### ISSUE THREAD

**Issue by nklincoln** [hyperledger/caliper#535]
At the moment the lifecycle of a caliper test cannot be easily controlled through the test config file.

It would be really nice if you could conditionally enable/disable the following via the config file:

- [x] start/end commands (noStart/noEnd)
- [ ] init (noConfigure)
- [ ] install smart contracts (noDeploy)
- [ ] only perform the init phase (initOnly)

With these additional flags, it would be nice if `caliper-flow` was modified to use them at the same time being converted into an actual Class for easier consumption by  the CLI (or any other user).

For instance, it might be that we create a new `CaliperEngine.js` class (or similar) that is interacted with by the caliper flow.js script to perform the appropriate flow control on the object

**Comment by aklenik** [comment]
The first item is already implemented: --caliper-skipstart/endscript

**Comment by nklincoln** [comment]
I think it would be nice for a user to be able to use either command line flags, or explicit config settings in a test config file. 

For instance, in addition to picking up any command line flags, we could introduce an optional section in the test config file under the key `flow`:

```
---
flow:
    skipStart: true/false
    skipInit: true/false
    skipInstall: true/false
    skipEnd: true/false
    testOnly: true/false
test:
  ~
monitor:
  ~
```

Then in `caliper-flow.js`, we could condition each key stage with 

```
  if (skipStart || (configObject.flow && configObject.flow.skipStart))  {
      logger.info('Skipping start commands due to passed flag');
  } else {
      doTheThing()
  }
```

and any form of 'testOnly/initOnly' can be used to define/override the appropriate 'skip' flags

**Comment by aklenik** [comment]
Closed by #556
